Sources/Modules/Yggdrasil/Furnace/Items/Accessories/HeatEmblem.cs

```diff
@@ -0,0 +1,37 @@
+using Terraria.Enums;
+
+namespace Everglow.Yggdrasil.Furnace.Items.Accessories;
+
+public class HeatEmblem : ModItem
+{
+	public const float BuffTriggerRate = 0.33f;
+	public const int BuffDuration = 300;
+
+	public override void SetDefaults()
+	{
+		Item.width = 44;
+		Item.height = 46;
+		Item.accessory = true;
+		Item.SetShopValues(ItemRarityColor.Blue1, Item.buyPrice(gold: 1, silver: 50));
+	}
+
+	public override void UpdateAccessory(Player player, bool hideVisual)
+	{
+		// 1. + 2/s Life Regeneration
+		player.lifeRegen += 4;
+
+		// 2. Gain "On Fire" debuff when get hurt
+		player.GetModPlayer<HeatEmblemPlayer>();
```

Comment:
ModPlayer is an autoloading class, which means to take effects when mod was loaded, instead of the frame when you invoke GetModPlayer<>.

Possible code

```
public class HeatEmblemPlayer : ModPlayer
{
    public bool HasHeatEmblem = false;
    public override void ResetEffects() {...}
	public override void OnHurt(Player.HurtInfo info)
	{
        if ( !HasHeatEmblem ) return false.
		if (Main.rand.NextFloat() < HeatEmblem.BuffTriggerRate)
		{
			Player.AddBuff(BuffID.OnFire, HeatEmblem.BuffDuration);
		}
	}
}
```

Reply:
Fixed